test/karma/behaviorSpec.js

```diff
@@ -0,0 +1,522 @@
+    });
+
+    it('attachTrackedViews after view.updateDOM', function() {
+      var methodsCalled = [];
+      var BehaviorTrackingAttachTrackedViews = TorsoBehavior.extend({
+        attachTrackedViews: function() {
+          methodsCalled.push('behavior:attachTrackedViews');
+        }
+      });
+      var ViewTrackingUpdateDom = TorsoView.extend({
+        behaviors: {
+          behaviorTrackingAttachTrackedViews: {
+            behavior: BehaviorTrackingAttachTrackedViews
+          }
+        },
+        updateDOM: function() {
+          methodsCalled.push('view:updateDOM');
+        }
+      });
+
+      var viewTrackingUpdateDom = new ViewTrackingUpdateDom();
+      viewTrackingUpdateDom.render();
+
+      expect(methodsCalled[0]).toBe('view:updateDOM');
+      expect(methodsCalled[1]).toBe('behavior:attachTrackedViews');
+    });
+
+
+
+    it('_attached before view._attached', function() {
+      var methodsCalled = [];
+      var ViewTrackingAttached = generateViewAndBehaviorTracking(methodsCalled, '_attached');
+      var viewTrackingAttached = new ViewTrackingAttached();
+
+      viewTrackingAttached.attachTo(this.$app);
+      expect(methodsCalled[0]).toBe('behavior:_attached');
+      expect(methodsCalled[1]).toBe('view:_attached');
+    });
+
+    it('_detached before view._detached', function() {
+      var methodsCalled = [];
+      var ViewTrackingDetached = generateViewAndBehaviorTracking(methodsCalled, '_detached');
+      var viewTrackingDetached = new ViewTrackingDetached();
+
+      viewTrackingDetached.attachTo(this.$app);
+      viewTrackingDetached.detach();
+
+      expect(methodsCalled[0]).toBe('behavior:_detached');
+      expect(methodsCalled[1]).toBe('view:_detached');
+    });
+
+    it('_activate before view._activate', function() {
+      var methodsCalled = [];
+      var ViewTrackingActivate = generateViewAndBehaviorTracking(methodsCalled, '_activate');
+      // Views are activated upon instantiation
+      var viewTrackingActivate = new ViewTrackingActivate();
+
+      expect(methodsCalled[0]).toBe('behavior:_activate');
+      expect(methodsCalled[1]).toBe('view:_activate');
+    });
+
+    it('_deactivate before view._deactivate', function() {
+      var methodsCalled = [];
+      var ViewTrackingDeactivate = generateViewAndBehaviorTracking(methodsCalled, '_deactivate');
+      var viewTrackingDeactivate = new ViewTrackingDeactivate();
+      viewTrackingDeactivate.deactivate();
+
+      expect(methodsCalled[0]).toBe('behavior:_deactivate');
+      expect(methodsCalled[1]).toBe('view:_deactivate');
+    });
+
+    it('_dispose before view._dispose', function() {
+      var methodsCalled = [];
+      var ViewTrackingDispose = generateViewAndBehaviorTracking(methodsCalled, '_dispose');
+      var viewTrackingDispose = new ViewTrackingDispose();
+      viewTrackingDispose.dispose();
+
+      expect(methodsCalled[0]).toBe('behavior:_dispose');
+      expect(methodsCalled[1]).toBe('view:_dispose');
+
```

Comment:
I could write the tests using auto-generated classes and custom matchers. The advantage is that you'd have less test boilerplate and better error messages, but it's harder to debug. Thoughts?